OpenAI-DotNet/Assistants/AssistantsEndpoint.cs

```diff
@@ -21,7 +22,7 @@ public sealed class AssistantsEndpoint : BaseEndPoint
         /// <returns><see cref="ListResponse{Assistant}"/></returns>
         public async Task<ListResponse<AssistantResponse>> ListAssistantsAsync(ListQuery query = null, CancellationToken cancellationToken = default)
         {
-            var response = await client.Client.GetAsync(GetUrl(queryParameters: query), cancellationToken).ConfigureAwait(false);
+            var response = await client.Client.GetResponseAsync(GetUrl(queryParameters: query), HttpMethod.Get, cancellationToken: cancellationToken);
```

Comment:
What is the advantage of changing this?

Reply:
`GetResponseAsync` is basically the wrapper/method that handles the whole new feature.
Every single GetAsync/GetPost/SendAsync/etc needs to go through `GetResponseAsync` for proper error handling.

It's supposed to represent the [following method in the python library](https://github.com/openai/openai-python/blob/e36956673d9049713c91bca6ce7aebe58638f483/src/openai/_base_client.py#L864), where they've chosen to represent internal http exceptions as user-friendly ones, as defined by their own library.

The method handles both 2 internal exceptions, as well as successfully received responses with status codes 4XX-5XX.

This is how our [own](https://github.com/kikaragyozov/OpenAI-DotNet/blob/7e43e9a442ce819d205041b2e431b2324f8a1adb/OpenAI-DotNet/Extensions/HttpClientExtensions.cs#L26) version looks for now.

[ThrowApiExceptionIfUnsuccessfulAsync](https://github.com/kikaragyozov/OpenAI-DotNet/blob/7e43e9a442ce819d205041b2e431b2324f8a1adb/OpenAI-DotNet/Extensions/ResponseExtensions.cs#L77) is called from within which handles the rest of the logic for 4XX-5XX status codes. 

I see why they did this - the consumer shouldn't be dealing with exceptions at such a low level, though in the .NET world perhaps the `OpenAITimeoutException` is unnecessary as anyone dealing with `CancellationTokens` would expect to receive a `TaskCanceledException` or an exception deriving of `OperationCanceledException`